### PR TITLE
Fix libroach tests on macOS

### PR DESCRIPTION
**Pull request by benesch**
See commit messages for details.

### CODE REVIEW

**Review by bdarnell — APPROVED** [review]
:lgtm: 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/27866#-:-LI8PKuh5kXAcrOezHYd:bg7igkv)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)

File: c-deps/libroach/testutils.cc [hunk 1]
```diff
@@ -30,38 +31,49 @@ void __attribute__((weak)) rocksDBLog(char*, int) {}
...
-    tmp_dir_ = std::string(tmp_c_dir);
-  } else {
-    std::cerr << "Error creating temp directory" << std::endl;
+  if (mkdtemp(tmpl) == NULL) {
+    err(1, "creating temporary directory %s", tmpl);
   }
-
-  delete[] tmpl;
-  return (tmp_c_dir != NULL);
+  tmp_dir_ = tmpl;
```
Comment (mberhault) [inline c-deps/libroach/testutils.cc]: You still need to delete `tmpl`, `tmp_dir_` is a basic_string that will hold a copy of the contents of `tmpl` up to the null character.

**Review by mberhault — APPROVED** [review]
LGTM, just a few tweaks. I'm not sure why the cleanup failed in TC, I patched it in and tried with both normal `make` and `builder.sh` and it's perfectly happy.

File: c-deps/libroach/ccl/db.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
+// Copyright 2018 The Cockroach Authors.
```
[the remark this answers is not in the document]
Comment (benesch) [reply c-deps/libroach/ccl/db.h]: D'oh. Thanks. Fixed.

File: c-deps/libroach/ccl/db_test.cc [hunk 3]
```diff
@@ -14,34 +14,42 @@
...
 using namespace testutils;
 
 namespace enginepbccl = cockroach::ccl::storageccl::engineccl::enginepbccl;
 
-TEST(LibroachCCL, DBOpenHook) {
+#include <libroach.h>
+#include "db.h"
+class CCLTest : public testing::Test {
+ protected:
+  void SetUp() override { DBSetOpenHook((void*)DBOpenHookCCL); }
```
[the remark this answers is not in the document]
Comment (benesch) [reply c-deps/libroach/ccl/db_test.cc]: Heh, good point. Done.

File: c-deps/libroach/testutils.cc (hunk 1 above)
Comment (benesch) [reply c-deps/libroach/testutils.cc]: Oops, thanks. Fixed.

**Comment by benesch** [comment]
Looks like passing `FTW_MOUNT` was too conservative. The TeamCity machines are configured such that files in the tempdir appear on a separate filesystem. I've removed that flag and everything looks AOK. Thanks for the reviews!

bors r=bdarnell,mberhault

**Pull request merged**